include/seqan3/alignment/scoring/detail/simd_matrix_scoring_scheme.hpp

```diff
@@ -111,59 +117,64 @@ class simd_matrix_scoring_scheme
      *\{
      */
     /*!\brief Given two simd vectors, compute an element-wise score and return another simd vector.
-     * \param[in] lhs The left operand to compare.
-     * \param[in] rhs The right operand to compare.
-     * \returns A simd vector of the same type as the input, with scores computed between both inputs.
+     * \param[in] score_profile The precomputed score profile.
+     * \param[in] ranks A simd vector over alphabet ranks.
+     * \returns A score simd vector computed based on the given score profile and alphabet ranks.
      *
      * ### Exception
      *
      * No-throw guarantee.
      *
      * ### Complexity
      *
-     * Linear in the length of one input vector (`lhs` and `rhs` are equally sized).
+     * Linear in the length of one input vector (`score_profile` and `ranks` are equally sized).
      *
      * ### Thread safety
      *
      * Thread-safe.
+     *
+     * \attention You need to call seqan3::detail::simd_matrix_scoring_scheme::make_score_profile before invoking
+     *            the score interface to convert the column alphabet ranks into a column profile. Not doing this cannot
+     *            be detected by the program and can lead to wrong results.
```

Comment:
Btw, since I need to handle the special case for 8bit scores, there is going to be a specific profle type I guess. This way it cannot be called wrongly. But for now, I keep it like this, as it is merely detail code.